scipy/ndimage/tests/test_filters.py

```diff
@@ -107,25 +129,25 @@ def _validate_complex(self, array, kernel, type2, mode='reflect', cval=0):
         # test correlate output dtype
         output = correlate(array, kernel, output=type2)
         assert_array_almost_equal(expected, output)
-        assert_equal(output.dtype.type, type2)
+        assert output.dtype.type == type2
 
         # test correlate with pre-allocated output
-        output = np.zeros_like(array, dtype=type2)
+        output = xp.zeros_like(array, dtype=type2)
         correlate(array, kernel, output=output)
         assert_array_almost_equal(expected, output)
```

Comment:
Does `assert_array_almost_equal` work for CuPy arrays?

Reply:
No. This is a shim from _lib.